### PR TITLE
feat: state of charge example

### PR DESCRIPTION
**Pull request by lsmienk**
In this example you can set the state of charge (SOC) and see how that influences the route.

### CODE REVIEW

File: state-of-charge/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { getDurationString } from '../utils';
+
+/**
+ * Draw a route on a map.
+ *
+ * The route object contains `polyline` data -  the polyline encoded route (series of coordinates as a single string).
+ * We need to decode this information first. We use Mapbox polyline tool (https://www.npmjs.com/package/@mapbox/polyline) for this.
+ * As a result of decoding we get pairs [latitude, longitude].
+ * To draw a route on a map we use Mapbox GL JS. This tool uses the format [longitude,latitude],
+ * so we have to reverse every pair.
```
Comment (timkuilman) [inline state-of-charge/index.js:14]: Seems like this is doubled? See a few lines further.

File: public/state-of-charge/index.html [hunk 2]
```diff
@@ -0,0 +1,66 @@
+<!DOCTYPE html>
+<html lang="nl">
+  <head>
+    <title>Chargetrip API: state of chargeexample</title>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <meta name="description" content="Build a route with the Chargetrip API" />
```
Comment (timkuilman) [inline public/state-of-charge/index.html:6]: Typo + description can be clearer.

File: public/state-of-charge/index.html [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    <link href="https://fonts.googleapis.com/css2?family=Inter:wght@100;200;400;600&display=swap" rel="stylesheet" />
+    <link href="https://api.mapbox.com/mapbox-gl-js/v1.10.0/mapbox-gl.css" rel="stylesheet" />
+    <link rel="icon" href="../favicon-57.png" type="image/x-icon" />
+    <link rel="stylesheet" href="style.css" />
+  </head>
+
+  <body>
+    <div id="map"></div>
+    <div class="github-link">
+      <a href="https://github.com/chargetrip/examples/tree/master/route" title="github project">
```
Comment (timkuilman) [inline public/state-of-charge/index.html:18]: This link is not correct.

File: state-of-charge/client.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        client.executeSubscription(createRequest(routeUpdate, { id: routeId })),
+        subscribe(result => {
+          const { status, route } = result.data?.routeUpdatedById;
+
+          // You can keep listening to the route changes to update route information.
+          // For this example we want to only draw the initial route.
+          if (status === 'done' && route) {
+            document.getElementById('calculating').style.display = 'none';
+            unsubscribe();
+            callback(result.data?.routeUpdatedById?.route);
```
Comment (timkuilman) [inline state-of-charge/client.js:64]: I don't completely get what you are doing with the callback. Can you explain this?

File: state-of-charge/queries.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            { chargingPower: 150, standard: CHADEMO }
+          ]
+          climate: true
+          minPower: 43
+          numberOfPassengers: 1
+        }
+        routeRequest: {
+          origin: {
+            type: Feature
+            geometry: { type: Point, coordinates: [${currentLocation}] }
```
[the remark this answers is not in the document]
Comment (lsmienk) [reply state-of-charge/queries.js:37]: Ah yes the initial idea was to be able to click on other stations and calculate from there. But as we are not doing that I will change it back.

File: state-of-charge/queries.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
+import qql from 'graphql-tag';
+
+/*
+ * In this example we request a route from Amsterdam, Netherlands to Berlin, Germany
+ * The changing conditions are:
+ *   - full battery at Amsterdam, Germany
+ *   - no desired range at Berlin, Germany
+ *   - EV can charge at CHadMO changers
+ *   - should use climate (temperature and weather conditions)
+ *   - the EV driver can drive 40 km  less than the EV specs (specs is 440 km, custom range is 400 km)
```
[the remark this answers is not in the document]
Comment (lsmienk) [reply state-of-charge/queries.js:10]: This comment doesn't apply to this example at all. I removed it

File: state-of-charge/map.js [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+  points.push({
+    type: 'Feature',
+    properties: {
+      description: legs[0].rangeStart,
+      icon: 'location_big',
+    },
+    geometry: legs[0].origin?.geometry,
+  });
+
+  legs.map((leg, index) => {
```
[the remark this answers is not in the document]
Comment (lsmienk) [reply state-of-charge/map.js:151]: I am not sure what you mean? We are using legs to create a points array. Could you explain?

File: public/state-of-charge/index.html (hunk 2 above)
Comment (lsmienk) [reply public/state-of-charge/index.html:6]: I agree. I change it to: Chargetrip API: example of the impact of your SOC
But I am not sure if that is much better, do you have suggestions?

File: state-of-charge/map.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (timkuilman) [reply state-of-charge/map.js:152]: Map returns a new array with the same length. For(each) loops don't return anything they simply iterate over a collection. You just iterate so a for loop is the better option. 

What you could do is: 

const points = legs.map(...)

File: state-of-charge/client.js (hunk 4 above)
Comment (lsmienk) [reply state-of-charge/client.js:62]: We are returning the route here. We have done this in the same way as in the elevationPlot example.

File: state-of-charge/map.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (lsmienk) [reply state-of-charge/map.js]: I ended up going with a for loop

**Pull request merged**